core/node/da_dispatcher/src/da_dispatcher.rs

```diff
@@ -0,0 +1,196 @@
+                    .dispatch_blob(batch.l1_batch_number.0, batch.pubdata.clone())
+            })
+            .await
+            .with_context(|| {
+                format!(
+                    "failed to dispatch a blob with batch_number: {}, pubdata_len: {}",
+                    batch.l1_batch_number,
+                    batch.pubdata.len()
+                )
+            })?;
+            let dispatch_latency_duration = dispatch_latency.observe();
+
+            let sent_at =
+                NaiveDateTime::from_timestamp_millis(Utc::now().timestamp_millis()).unwrap();
+
+            let mut conn = pool.connection_tagged("da_dispatcher").await?;
+            conn.data_availability_dal()
+                .insert_l1_batch_da(
+                    batch.l1_batch_number,
+                    dispatch_response.blob_id.as_str(),
+                    sent_at,
+                )
+                .await?;
+            drop(conn);
+
+            METRICS
+                .last_dispatched_l1_batch
+                .set(batch.l1_batch_number.0 as usize);
+            METRICS.blob_size.observe(batch.pubdata.len());
+            tracing::info!(
+                "Dispatched a DA for batch_number: {}, pubdata_size: {}, dispatch_latency ms: {}",
+                batch.l1_batch_number,
+                batch.pubdata.len(),
+                dispatch_latency_duration.as_millis()
+            );
+        }
+
+        Ok(())
+    }
+
+    /// Polls the data availability layer for inclusion data, and saves it in the database.
+    async fn poll_for_inclusion(&self, pool: &ConnectionPool<Core>) -> anyhow::Result<()> {
+        let mut conn = pool.connection_tagged("da_dispatcher").await?;
+        if let Some(blob_info) = conn
+            .data_availability_dal()
+            .get_first_da_blob_awaiting_inclusion()
+            .await?
+        {
+            drop(conn);
+            let inclusion_data = self
+                .client
+                .get_inclusion_data(blob_info.blob_id.clone())
+                .await
+                .with_context(|| {
+                    format!(
+                        "failed to get inclusion data for blob_id: {}, batch_number: {}",
+                        blob_info.blob_id, blob_info.l1_batch_number
+                    )
+                })?;
+
+            let mut conn = pool.connection_tagged("da_dispatcher").await?;
+            if let Some(inclusion_data) = inclusion_data {
+                conn.data_availability_dal()
+                    .save_l1_batch_inclusion_data(
+                        L1BatchNumber(blob_info.l1_batch_number.0),
+                        inclusion_data.data.as_slice(),
+                    )
+                    .await?;
+                drop(conn);
+
+                let inclusion_latency = Utc::now().signed_duration_since(blob_info.sent_at);
+                METRICS
+                    .inclusion_latency
+                    .observe(inclusion_latency.to_std()?);
```

Comment:
Bikeshedding: I wouldn't return an error here, even given that errors are swallowed in the call chain above:

```rust
if let Ok(latency) = inclusion_latency.to_std() {
    METRICS.inclusion_latency.observe(latency);
}
```

Returning an error just because we cannot measure a metric looks weird.